### ISSUE TITLE
Some question about the model.

### ISSUE THREAD

**Issue by yzldw333** [browatbn2/3FabRec#6]
Hi, 
     it's a great project, and I run successfully!
     Can I know your demo's model trained from which dataset, and is it the best one you mentioned in your paper?

**Comment by browatbn2** [comment]
Hi, thank you! It's the pre-trained model as in the paper with landmarks trained on WFLW. It's without encoder finetuning, however, so the original autoencoder stays intact and it can be used to train landmarks on other datasets. 

There are a few more detail here:
https://github.com/browatbn2/3FabRec/issues/1#issue-636810283

**Comment by yzldw333** [comment]
So, It's trained on VGGFace2 and AffectNet for encoder and decoder, then, the Interleaved transfer layers finetuned on WFLW (7500 images), am I right?

**Comment by yzldw333** [comment]
Becase I tested some cases like eye close (single/double), the landmark accuracy seems not good, maybe it's because of the training dataset, and I wanted to know if you have one best model open for us.
Also, I have another question, When I read your paper, I thought it's like a transfer-learning work, have you try training WFLW use the present model without pretrained on VGGFace2 and AffectNet, which means all parameters can be trained for regression. It's interesting to see the accuracy result between them.

**Comment by browatbn2** [comment]
> So, It's trained on VGGFace2 and AffectNet for encoder and decoder, then, the Interleaved transfer layers finetuned on WFLW (7500 images), am I right?
Yup, exactly.

> Becase I tested some cases like eye close (single/double), the landmark accuracy seems not good, maybe it's because of the training dataset, and I wanted to know if you have one best model open for us.
What dataset did you test on? WFLW? The accuracy should be fine for this model too. I'll check and upload the other models but I probably won't get to it this week anymore. 

> have you try training WFLW use the present model without pretrained on VGGFace2 and AffectNet, which means all parameters can be trained for regression. It's interesting to see the accuracy result between them.
Yes, I did that for 300W, it's in the supplementary materials.

**Comment by yzldw333** [comment]
Sorry to bother you.
I saw your code and read "Robust Discrimination and Generation of Faces
using Compact, Disentangled Embeddings", doubt about Loss(enc), does it use losses described as in disentangled network? I saw only L1 regularization loss, is that true?

**Comment by browatbn2** [comment]
Hi, loss_enc actually doesn't use L1 regularization loss, the flag is set to False. So, yes, the autoencoder uses the same adversarial loss_enc as the disentanglement network. It might be a bit hidden in the code, have a look at train_aae_unsupervised.py:246 in function self.update_encoding().

**Comment by yzldw333** [comment]
Yes, I found it, maybe the adversarial losses here makes z more natural, permute op exists in with_gen_loss... and this was not used, right? 
and attribute supervised loss was also not used, the only loss for z was the adversarial loss which makes z the same distribution random noise.

**Comment by browatbn2** [comment]
Yes, no loss_gen and attribute supervision here. There are no attributes here, it's completely unsupervised at this point. 

Exactly, the adversarial encoding loss makes z normally distributed. This way we can sample from a Gaussian to generate new faces and it ensures that the landmarks that are learned later (from labelled samples) generalize well to unseen faces.

**Comment by browatbn2** [comment]
I finally had some time to make some updates and upload more pre-trained models. Since the code has changed a little since the publication, these models are not identical to the models used in the paper but the results are almost the same. 

Just run `./download_models.sh` to get them. 

There is now:
- Trained autoencoder w/o landmarks
- Landmark detector for 300W (NME=3.85 on full test set)
- Landmark detector for WFLW (NME=5.57 on full test set)
- Landmark detector for AFLW (NME=1.79 on full test set)
- Demo model
